lesson21/src/js/main.js

```diff
@@ -0,0 +1,213 @@
+import '../css/style.css'
+import * as loading from "./module/loading";
+import { createAttributedElements} from "./utils/createAttributedElements";
+
+const endPointURL = {
+    usersData: "https://myjson.dit.upm.es/api/bins/i55y"
+}
+
+const createErrorMessage = (error,element) => {
+    const errorMessage = createAttributedElements({
+        tag:"p",
+        valuesByAttributes:{
+            class:"error-message"
+        },
+        str:error
+    });
+    element.appendChild(errorMessage);
+}
+
+const fetchData = async (endpointURL,element) => {
+    const response = await fetch(endpointURL);
+    if(!response.ok){
+        console.error(`${response.status}:${response.statusText}`);
+        createErrorMessage("Communication with the server is broken.",element);
+        return;
+    }
+    const json = await response.json();
+    return json.data;
+}
+
+const fetchContentsData = (endPointURL,element,ms) => new Promise(resolve => setTimeout(() => resolve(fetchData(endPointURL,element)),ms));
+
+const table = document.getElementById("js-table");
+
+const initUsersData = async () => {
+    loading.showLoading(table);
+    try{
+        const usersData = await fetchContentsData(endPointURL.usersData,table,500);
+        if (!usersData.length) {
+            createErrorMessage("No user.",table);
+            return;
+        }
+        renderTableElements(usersData);
+        sortTableBody(usersData);
+    }catch(error){
+        console.error(error);
+    }finally{
+       loading.removeLoading(table);
+    }
+}
+
+const renderTableElements = usersData => {
+    renderTableHeader();
+    renderTableBody(usersData);
+}
+
+const usersTableColumn = {
+    "id"     : "ID",
+    "name"   : "名前",
+    "gender" : "性別",
+    "age"    : "年齢"
+}
+
+const renderTableHeader = () => {
+    const thead = document.createElement("thead");
+    const tr = document.createElement("tr");
+    thead.className = "bg-slate-500";
+    tr.className = "";
```

Comment:
Q.
Is it necessary?
I'm sorry I haven't learned tailwind css properly yet.

Reply:
Thank you for your comment!
This is an unneeded code forgotten. :scream:
Thank you for teaching!

I fixed.
https://github.com/nakagawamai/js-lesson/pull/23/commits/70a34f00ee0c0b6d976f57d3b4f5b686f23e8080